src/client/Ruby/ms-rest/lib/ms_rest/service_client.rb

```diff
@@ -20,7 +25,45 @@ class ServiceClient
     #
     def initialize(credentials, options = nil)
       @credentials = credentials
+      @request_headers = {}
+      @middlewares = {middlewares: [[MsRest::RetryPolicyMiddleware, times: 3, retry: 0.02], [:cookie_jar]]}
+    end
+
+    #
+    # @param base_url [String] the base url for the request.
+    # @param method [Symbol] with any of the following values :get, :put, :post, :patch, :delete.
+    # @param path [String] the path, relative to {base_url}.
+    # @param options [Hash{String=>String}] specifying any request options like :credentials, :body, etc.
+    # @return [Concurrent::Promise] Promise object which holds the HTTP response.
+    #
+    def make_request_async(base_url, method, path, options = {})
+      options = @middlewares.merge(options)
+      options[:credentials] = options[:credentials] || @credentials
+      request  = MsRest::HttpOperationRequest.new(base_url, path, method, options)
+      promise = request.run_promise do |req|
+        options[:credentials].sign_request(req) unless options[:credentials].nil?
+      end
+      promise = promise.then do |http_response|
+        response_content = http_response.body.to_s.empty? ? nil : http_response.body
+        # Create Result
+        result = create_response(request, http_response)
+        result.body = response_content
```

Comment:
Should we send `response_content` into the initializer of the `create_response` method to eliminate line number 50?